### ISSUE TITLE
nextjs 12.3.1 SSR deployment success but HTTP ERROR 404 

### ISSUE THREAD

**Issue by unnimona** [cloudflare/next-on-pages#18]
i have uploaded nextjs ssr with wrangler and also tried via other methods
nextjs version nextjs 12.3.1 
the upload succeeds but page the deployment page returns  HTTP ERROR 404 

how to fix this issue?

**Comment by unnimona** [comment]
need help

**Comment by arielvieira** [comment]
Have you set the `NODE_VERSION=14` and added the `transformstream_enable_standard_constructor` and `streams_enable_constructors` Compatibility flags in the cloudflare pages project Settings?

**Comment by Ganitzsh** [comment]
Experiencing the same issue right now. I'm using the latest version of nextjs `13.0.3` with the compatibility flags under the functions tab. My automatic redirect does not work and when I try to go to the url directly I still get a 404:

```
// next.config.js
const redirects = async () => {
  return [
    {
      source: '/',
      destination: '/en',
      permanent: true,
    },
  ];
};

module.exports = {
  redirects,
  experimental: {
    appDir: true,
    runtime: 'experimental-edge',
  },
  images: {
    remotePatterns: [
      {
        protocol: 'http',
        hostname: 'localhost',
        port: '1337',
        pathname: '/uploads/**',
      },
    ],
  },
};

```

**Comment by arielvieira** [comment]
@Ganitzsh For what I see so far, I couldn't get next.js redirects/rewrites working with `next-on-pages`, I think it is not support next.js.

**Comment by Ganitzsh** [comment]
@arielvieira yeah, but even a single page app with a blank page gives me 404, so I think something else is wrong. I'm deploying my website on Vercel now because `next-on-pages` doesn't work with version 13. It'd be nice if I could centralise it on Cloudflare for everything.

**Comment by codeoholic** [comment]
Adding a websocket connection inside useEffect also causes 404.

**Comment by clairegraham** [comment]
Same issue here. It works fine in local dev, but Pages says it deployed successfully and yet every route returns a 404.

Running ```Next 13.0.4```, ```Wrangler 2.4.4```, and ```next-on-pages 0.2.0```

Tried Vercel and it worked immediately without any changes.

**Comment by paulonteri** [comment]
> Same issue here. It works fine in local dev, but Pages says it deployed successfully and yet every route returns a 404.
> 
> Running `Next 13.0.4`, `Wrangler 2.4.4`, and `next-on-pages 0.2.0`
> 
> Tried Vercel and it worked immediately without any changes.

Same here

**Comment by Ganitzsh** [comment]
> Has anyone managed to fix this I'm having the same issue with 404

Unfortunately not, it seems this issue is being ghosted. It'd be nice to have at least some visibility on this 🙏

**Comment by sheamedia** [comment]
Bumping into this also. `Wrangler 2.4.4`, `NextJS 13.0.6`. New install using the /app dir. Getting 404 at / and /second. Has anyone found a workaround?

**Comment by sheamedia** [comment]
Hi everyone, I was able to get a `NextJS` app to load on Cloudflare Pages by installing `"next": 13.0.6` as a TypeScript app. `13.0.7` throws a `TypeError` so avoid the `@latest` for now. Everything else was set up conforming to the guidance in the README.md. I hope this helps others to move forward with their projects.

**Comment by sgoodluck** [comment]
> Hi everyone, I was able to get a `NextJS` app to load on Cloudflare Pages by installing `"next": 13.0.6` as a TypeScript app. `13.0.7` throws a `TypeError` so avoid the `@latest` for now. Everything else was set up conforming to the guidance in the README.md. I hope this helps others to move forward with their projects.

@EvanScoboria Are you doing full SWR or also leveraging static only pages? I can only seem to get this working when every page is streamed.

**Comment by 2-fly-4-ai** [comment]
Just use netlify. It's effortless. The hassle trying to get this setup on Cloudflare + the lack of support means it is not ready. Despite misleading blog posts by Cloudflare stating that it is.  

Got a free netlify account. CLick build. Done.

Everything working like it should. Even better than deploying on vercel, the Images load faster, You can disable the crazy prefetch behavior and your server function limit will stay very low. Put it on a cloudflare proxy and call it a day.

Kick back, enjoy your life instead of debugging half built undocumented trash.

**Comment by sgoodluck** [comment]
> Just use netlify. It's effortless. The hassle trying to get this setup on Cloudflare + the lack of support means it is not ready. Despite misleading blog posts by Cloudflare stating that it is.  
> 
> Got a free netlify account. CLick build. Done.
> 
> Everything working like it should. Even better than deploying on vercel, the Images load faster, You can disable the crazy prefetch behavior and your server function limit will stay very low. Put it on a cloudflare proxy and call it a day.
> 
> Kick back, enjoy your life instead of debugging half built undocumented trash. 

I understand the frustration. Calling the work trash however isn't professional nor is it useful. So please refrain from that.

Agree that Netlify and Vercel are a breeze deployment wise. At scale they do become quite expensive comparatively though. So there are tradeoffs.

I managed to get everything up and running on cloudflare workers yesterday. I'll look into making a PR to document that process so it's less painful for others to get to a point of workability deploying directly via cloudflare or via GitHub actions.

**Comment by EvanScoboria** [comment]
@sgoodluck Honestly, I ran into so many bugs that I also switched over to Vercel for now. I'd be very interested to read through your PR when posted though. I'm excited for this to work consistently.

**Comment by 2-fly-4-ai** [comment]
> Just use netlify. It's effortless. The hassle of trying to get this set up on Cloudflare + the lack of support means it is not ready. Despite misleading blog posts by Cloudflare stating that it is.
> 
> Got a free netlify account. CLick build. Done.
> 
> Everything working like it should. Even better than deploying on vercel, the Images load faster, You can disable the crazy prefetch behavior and your server function limit will stay very low. Put it on a cloudflare proxy and call it a day.
> 
> Kick back, enjoy your life instead of debugging half-built undocumented trash.

If I can emphasize how painful the process is, maybe my use of a BAD word like trash is still useful to someone.    
On the other hand, they should not advertise a product as Available and ready to use when it clearly is not. 

I consulted with people in the Cloudflare discord, the admins even made a separate channel and basically said, "sorry, there is no support for this at this time". I came across a bunch of people who had also tried to deploy with 0 success. What do you think the sum of the total wasted hours by all these developers is? 

We are all about Cloudflare. Who doesn't like cheap infrastructure that scales well? Do you think this is ready to deploy"At scale" in its current state? This project is unsupported and not ready. Or requires a level of expertise above most who have attempted it. For most people attempting this, it is going to be a Huge waste of time. 

If you do this : 

"look into making a PR to document that process so it's less painful for others to get to a point of workability deploying directly via cloudflare or via GitHub actions."

Please share it and recommend that Either Greg Updates his official post @cloudflare https://blog.cloudflare.com/next-on-pages/ or you amend his article to include your results. There are seasoned developers here struggling with its implementation, it is totally out of reach of the majority of people that deploy next.  

Fewer users mean fewer contributors and fewer people invested in making it work. This means even if it is open source, it's dead and dying before it's even taken off. It's not free after all, you still need to pay for serverless functions. If they actually made this work there would have been an exodus of people from those other platforms to host on Cloudflare. No one likes dealing with excessive charges, paying for image optimizations, and other restrictive practices.

**Comment by paulonteri** [comment]
> making a PR

Hi, were you able to make the PR?

**Comment by paulonteri** [comment]
> I managed to get everything up and running on cloudflare workers yesterday. I'll look into making a PR to document that process so it's less painful for others to get to a point of workability deploying directly via cloudflare or via GitHub actions.

Hi @sgoodluck, were you able to make the PR?

**Comment by james-elicx** [comment]
> Hi @sgoodluck, were you able to make the PR?

Have you tried the latest version of next-on-pages and nextjs @paulonteri ? There have been many bug fixes in the past month that might have addressed this.

**Comment by sgoodluck** [comment]
> > I managed to get everything up and running on cloudflare workers yesterday. I'll look into making a PR to document that process so it's less painful for others to get to a point of workability deploying directly via cloudflare or via GitHub actions.
> 
> Hi @sgoodluck, were you able to make the PR?

I have not yet! But I still can if the latest that @james-elicx mentioned doesn't handle things.
Overall, the deployment works fine but you need to be intentional about how things are deployed and the project is structured.

Could you try the latest? 
If that doesn't work, I'll fork off a skeleton project that deploys as a starting point. 

My main concern is that my setup may be version specific and not-interop-friendly when updates come through!
